### PR TITLE
LPS-162178 | master

### PR DESCRIPTION
**Pull request by chaowang-123**
https://issues.liferay.com/browse/LPS-162178

### CODE REVIEW

**Comment by chaowang-123** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 1]
```diff
@@ -1880,6 +1880,79 @@ definition {
 		}
 	}
 
+	@description = "LPS-162178 - Verify that the user is able to use the ERC field in Groovy Scripts, on the Validations tab"
+	@priority = "5"
+	test CanUseERCFieldWithGroovyScript {
```
Comment (epark18) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1885]: Could you add this test to acceptance?

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase [hunk 2]
```diff
@@ -1880,6 +1880,79 @@ definition {
...
+				validationType = "Groovy");
+
+			ObjectCustomValidation.gotoCustomValidation(validationName = "Custom Validation");
+
+			ObjectCustomValidation.activeValidation();
+
+			ObjectCustomValidation.gotoTab(tabName = "Conditions");
+
+			ObjectCustomValidation.insertScriptOnValidation(scriptValidation = "invalidFields = (customObjectField == \"Invalid Entry\")");
+
```
Comment (epark18) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectFields.testcase:1921]: The validation needs to use the externalReferenceCode. This is a good start, but you can change Invalid Entry to use ERC instead.

**Review by epark18 — COMMENTED** [review]
Hey @chaowang-123, this test is good, but the main change you need to make is adding ERC to the validation. Something to note is that ERC in the validation references the ERC of the object Entry and not the object definition. I think the easiest way to validate this test would be to:
1. Create an object with field
2. Give it a validation using ERC like the one mentioned in the comment.
3. Keep your Custom Views section so you can view the ERC of an object Entry.
4. Create an Object Entry.
5. Identify it's ERC and edit that same Entry to use its ERC as the new Entry.
6. Assert the error in your validation is triggered.
Let me know if you have any questions. Thanks!

**Comment by chaowang-123** [comment]
ci:test:sf

**Comment by epark18** [comment]
@chaowang-123 Very nice! I really like the macro change as well. Thank you for this.

**Comment by epark18** [comment]
ci:forward

**Pull request closed without merging**